identity/src/main/java/com/android/identity/CborUtil.java

```diff
@@ -0,0 +1,305 @@
+
+        boolean pretty = ((flags & DIAGNOSTICS_FLAG_PRETTY_PRINT) != 0);
+        String indentString = "";
+        if (pretty) {
+            StringBuilder indentBuilder = new StringBuilder();
+            for (int n = 0; n < indent; n++) {
+                indentBuilder.append(' ');
+            }
+            indentString = indentBuilder.toString();
+        }
+
+        if (dataItem.hasTag()) {
+            sb.append(String.format(Locale.US, "%d(", dataItem.getTag().getValue()));
+        }
+
+        switch (dataItem.getMajorType()) {
+            case INVALID:
+                sb.append("<invalid>");
+                break;
+
+            case UNSIGNED_INTEGER:
+                // Major type 0: an unsigned integer.
+                sb.append(((UnsignedInteger) dataItem).getValue());
+                break;
+
+            case NEGATIVE_INTEGER:
+                // Major type 1: a negative integer.
+                sb.append(((NegativeInteger) dataItem).getValue());
+                break;
+
+            case BYTE_STRING:
+                // Major type 2: a byte string.
+                byte[] bstrValue = ((ByteString) dataItem).getBytes();
+
+                if (dataItem.hasTag() && dataItem.getTag().getValue() == 24
+                        && (flags & DIAGNOSTICS_FLAG_EMBEDDED_CBOR) != 0) {
+                    sb.append("<< ");
+                    ByteArrayInputStream bais = new ByteArrayInputStream(bstrValue);
+                    List<DataItem> dataItems = null;
+                    try {
+                        dataItems = new CborDecoder(bais).decode();
+                        if (dataItems.size() >= 1) {
+                            if (dataItems.size() > 1) {
+                                Logger.w(TAG, "Multiple data items in embedded CBOR, "
+                                        + "only printing the first");
+                            }
+                            toDiagnostics(sb, indent, Util.cborDecode(bstrValue), flags);
+                        } else {
+                            sb.append("Error: 0 Data Items");
+                        }
+                    } catch (CborException e) {
+                        // Never throw an exception
+                        sb.append("Error Decoding CBOR");
+                        Logger.w(TAG, "Error decoding CBOR: " + e);
+                    }
+                    sb.append(" >>");
+                } else {
+                    sb.append("h'");
+                    count = 0;
+                    for (byte b : bstrValue) {
+                        sb.append(String.format("%02x", b));
+                        count++;
+                    }
+                    sb.append("'");
+                }
+                break;
+
+            case UNICODE_STRING:
+                // Major type 3: string of Unicode characters that is encoded as UTF-8 [RFC3629].
+                String strValue = Util.checkedStringValue(dataItem);
+                // TODO: escape " in |value|
```

Comment:
Why?

Reply:
Why escaping? Because that's how it works, see for example cbor.me with the string **foobar " foobar**. I removed the TODO by actually just doing the escape.

<img width="918" alt="cbor-me-escape-double-quotes" src="https://user-images.githubusercontent.com/5580198/210648246-ab90ad0a-c9d8-4348-b204-702bf3a4632d.png">